packages/block-library/src/navigation/edit/use-generated-slug.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+function useGeneratedSlug( clientId ) {
```

Comment:
@draganescu Do you agree that we should remove this logic from this PR and just keep it simple and allow the REST API (via WP Core) to determine the slug for us?